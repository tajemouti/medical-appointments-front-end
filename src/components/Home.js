import NavigationBar from './NavigationBar';

function Home() {
  return (
    <>
      <div className='flex flex-row w-[100dvw]'>
        <div className='flex w-[15%]'>
          <NavigationBar />
        </div>
        <div className='flex w-[85%] bg-white'>
          <h1>home component</h1>
        </div>
      </div>
    </>
  );
}

export default Home;
