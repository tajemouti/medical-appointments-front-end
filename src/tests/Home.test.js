import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '../components/Home';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}));

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

describe('Tests home component', () => {
  it('should render page', () => {
    const homeComponent = renderer.create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    expect(homeComponent).toMatchSnapshot();
  });

  it('should display navigation buttons "Doctors"', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    const button1 = screen.getByText('Delete Doctor');
    expect(button1).toBeInTheDocument();
  });

  it('should display navigation buttons "Add Doctor"', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    const button1 = screen.getByText('Add Doctor');
    expect(button1).toBeInTheDocument();
  });

  it('should display navigation buttons "Doctors"', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    const button1 = screen.getByText('Doctors');
    expect(button1).toBeInTheDocument();
  });

  it('should display navigation buttons "My Appointments"', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    const button1 = screen.getByText('My Appointments');
    expect(button1).toBeInTheDocument();
  });

  it('should display navigation buttons "Book Appointment"', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    const button1 = screen.getByText('Book Appointment');
    expect(button1).toBeInTheDocument();
  });
});
