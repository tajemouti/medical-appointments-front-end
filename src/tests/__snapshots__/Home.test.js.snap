// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests home component should render page 1`] = `
<div
  className="flex flex-row h-[100dvh] justify-center md:w-[100dvw] md:flex md:flex-row"
>
  <div
    className="md:flex md:w-[15%]"
  >
    <button
      className="md:hidden absolute top-4 left-4 mr-4 p-2 text-gray-500 hover:text-gray-700 focus:outline-none f ocus:ring focus:ring-gray-200"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
    <div
      className="hidden md:flex flex-col pt-4 pb-12 md:pb-1 justify-between items-center h-[100dvh] bg-white  w-full drop-shadow-md"
    >
      <div>
        <img
          alt=""
          src="logo.svg"
        />
        <button
          className="md:hidden absolute top-4 left-4 mr-4 p-2 text-gray-500 hover:text-gray-700 focus:outline-none f ocus:ring focus:ring-gray-200"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </button>
      </div>
      <nav
        className="w-full"
      >
        <ul
          className="flex flex-col"
        >
          <a
            aria-current="page"
            className="active-link active"
            href="/"
            onClick={[Function]}
          >
            <li
              className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
            >
              Doctors
            </li>
          </a>
          <a
            className="active-link"
            href="/bookappointment"
            onClick={[Function]}
          >
            <li
              className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
            >
              Book Appointment
            </li>
          </a>
          <a
            className="active-link"
            href="/myappointment"
            onClick={[Function]}
          >
            <li
              className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
            >
              My Appointments
            </li>
          </a>
          <a
            className="active-link"
            href="/add-doctor"
            onClick={[Function]}
          >
            <li
              className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
            >
              Add Doctor
            </li>
          </a>
          <a
            className="active-link"
            href="/delete-doctor"
            onClick={[Function]}
          >
            <li
              className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
            >
              Delete Doctor
            </li>
          </a>
          <li
            className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
          >
            <button
              className="focus:outline-none"
              onClick={[Function]}
              type="button"
            >
              Logout
            </button>
          </li>
        </ul>
      </nav>
      <footer
        className="flex flex-col gap-4 w-full pt-4 pb-2 pl-4 pr-4"
      >
        <ul
          className=" justify-start items-end gap-3.5 inline-flex"
        >
          <li
            className="w-6 h-6 relative"
          >
            <img
              alt=""
              src="fb.svg"
            />
          </li>
          <li
            className="w-6 h-6 relative"
          >
            <img
              alt=""
              src="x.svg"
            />
          </li>
          <li
            className="w-6 h-6 relative"
          >
            <img
              alt=""
              src="google.svg"
            />
          </li>
          <li
            className="w-6 h-6 relative"
          >
            <img
              alt=""
              src="github.svg"
            />
          </li>
          <li
            className="w-6 h-6 relative"
          >
            <img
              alt=""
              src="linkedin.svg"
            />
          </li>
        </ul>
        <p
          className="text-gray-300 text-xs font-medium font-['Inter'] leading-[18px] tracking-[2.88px]"
        >
          @ 2023 Health Clinic
        </p>
      </footer>
    </div>
  </div>
  <div
    className="flex flex-col md:w-[85%] w-[100%] bg-white justify-center items-center "
  >
    <h1
      className="text-slate-800 text-4xl font-bold font-['Inter'] leading-[44px] mb-8"
    >
      Our Doctors
    </h1>
    <div
      className="flex flex-col md:flex md:flex-row md:justify-center md:items-center gap-4"
    />
    <div
      className="flex justify-between w-[85%] mt-4 absolute"
    >
      <button
        aria-label="Previous"
        className="hidden md:block md:w-[114px] md:h-[74px] md:rounded-r-[80px] bg-gray-300 "
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-white css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowBackIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </button>
      <button
        aria-label="Next"
        className="hidden md:block md:w-[114px] md:h-[74px] md:rounded-l-[80px] bg-gray-300"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-white css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowForwardIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
